src/test/java/marquez/api/resources/JobResourceTest.java

```diff
@@ -322,6 +326,69 @@ public void testJobRunCompleteWithUnparseableRunId() throws Exception {
     assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), res.getStatus());
   }
 
+  @Test
+  public void testGetAllRunsOfJob() throws MarquezServiceException {
+    JobResponse job = generateApiJob();
+    List<JobRun> jobRuns = Arrays.asList(Generator.genJobRun(), Generator.genJobRun());
+    when(MOCK_NAMESPACE_SERVICE.exists(NAMESPACE_NAME)).thenReturn(true);
+    when(MOCK_JOB_SERVICE.getAllRunsOfJob(
+            NAMESPACE_NAME, job.getName(), DEFAULT_LIMIT, DEFAULT_OFFSET))
+        .thenReturn(jobRuns);
+
+    Response response =
+        JOB_RESOURCE.getRunsForJob(NAMESPACE_NAME, job.getName(), DEFAULT_LIMIT, DEFAULT_OFFSET);
+
+    List<JobRunResponse> responseJobRuns = new ArrayList<JobRunResponse>();
+    for (Object resItem : (List<?>) response.getEntity()) {
+      responseJobRuns.add((JobRunResponse) resItem);
+    }
+    assertEquals(jobRuns.size(), responseJobRuns.size());
+  }
+
+  @Test
+  public void testGetAllRunsOfJob_namespaceNotFound() throws MarquezServiceException {
+    when(MOCK_NAMESPACE_SERVICE.exists(NAMESPACE_NAME)).thenReturn(false);
+
+    Response response =
+        JOB_RESOURCE.getRunsForJob(
+            NAMESPACE_NAME, "nonexistent_job", DEFAULT_LIMIT, DEFAULT_OFFSET);
+
+    assertEquals(Response.Status.NOT_FOUND.getStatusCode(), response.getStatus());
+  }
+
+  @Test
+  public void testGetAllRunsOfJob_noJobRuns() throws MarquezServiceException {
```

Comment:
If no job runs are found for a given job, I think I'd be more ideal to return an empty set of runs rather then a `404`?

```
{
  "runs": []
}
```

Reply:
fixed